lib/compress/zstd_fast.c

```diff
@@ -73,26 +96,63 @@ size_t ZSTD_compressBlock_fast_generic(
         U32 const current = (U32)(ip-base);
         U32 const matchIndex = hashTable[h];
         const BYTE* match = base + matchIndex;
+        const ptrdiff_t repIndex = (ptrdiff_t)current + 1 - offset_1;
+        const BYTE* repBase = (hasDict == ZSTD_dictMatchState
```

Comment:
Heads up, I tried a similar strategy for merging single-segment and ext-dict compressors, and it worked for `ZSTD_fast`, but I couldn't get it to work on `ZSTD_doubleFast`. You might have better luck, but it is something to be aware of.